### ISSUE TITLE
Check args[0] twice in xxtea.cc Handle<Value> xxtea

### ISSUE THREAD

**Issue by xiyuera** [Foreverflying/xxtea_c_4node#1]
line 16

``` c
    if( args.Length() != 2 || !Buffer::HasInstance( args[0] ) || !Buffer::HasInstance( args[0] ) ){
        ThrowException( Exception::TypeError( String::New( "Wrong arguments" ) ) );
        return scope.Close( Undefined() );
    }
```

**Comment by xiyuera** [comment]
This is the version for 0.12. tested.

``` cpp
#include <node.h>
#include <v8.h>
#include <node_buffer.h>
#include <stdlib.h>
#include <string.h>
extern "C" {
#include "xxtea.h"
}

using namespace v8;
using namespace node;

void xxtea( const FunctionCallbackInfo<Value>& args, bool encrypt )
{
    Isolate* isolate = Isolate::GetCurrent();
    HandleScope scope(isolate);

    if( args.Length() != 2 || !Buffer::HasInstance( args[0] ) || !Buffer::HasInstance( args[1] ) ){
        isolate->ThrowException(Exception::TypeError(
        String::NewFromUtf8(isolate, "Wrong arguments")));

        return;
    }

    char* plainBuf = Buffer::Data( args[0] );
    size_t plainLen = Buffer::Length( args[0] );
    char* keyBuf = Buffer::Data( args[1] );
    size_t keyLen = Buffer::Length( args[1] );
    unsigned char* ret;
    xxtea_long retLen;
    if( encrypt ){
        ret = xxtea_encrypt( (unsigned char*)plainBuf, (xxtea_long)plainLen, (unsigned char*)keyBuf, (xxtea_long)keyLen, &retLen );
    }else{
        ret = xxtea_decrypt( (unsigned char*)plainBuf, (xxtea_long)plainLen, (unsigned char*)keyBuf, (xxtea_long)keyLen, &retLen );
    }
    v8::Local<v8::Object> slowBuffer = Buffer::New( isolate, (const char*)ret, retLen );
    free( ret );

    args.GetReturnValue().Set(slowBuffer);
}

void encrypt( const FunctionCallbackInfo<Value>& args )
{
    return xxtea( args, true );
}

void decrypt( const FunctionCallbackInfo<Value>& args )
{
    return xxtea( args, false );
}

void init( Handle<Object> target )
{
    NODE_SET_METHOD( target, "encrypt", encrypt );
    NODE_SET_METHOD( target, "decrypt", decrypt );
}

NODE_MODULE( xxtea_c, init );
```

**Comment by foreverflying** [comment]
thank you!
bug fixed, new version is 0.1.6, you can use npm install it.
